File: GreenSignal/Data/Models/Petition.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.Models
{
    public   class Petition
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string SerialNumber { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }

        [Required]
        public Guid DepartmentId { get; set; }
        public Department Department { get; set; }

        public Guid? IncidentReportId { get; set; }
        public IncidentReport? IncidentReport { get; set; }

        public Guid? ParentPetitionId { get; set; }
        public Petition? ParentPetition { get; set; }

        [Required]
        public Guid InspectorId { get; set; }
        public Inspector Inspector { get; set; }

        [Required]
        public PetitionStatus Status { get; set; }

        [Required]
        public PetitionKind Kind { get; set; }

        [Required]
        public int AttributeVersion { get; set; }

        public IEnumerable<PetitionAttachment> Attachments { get; set; }
        public IEnumerable<PetitionAttribute> Attributes { get; set; }
        public IEnumerable<ReceiveMessage> ReceiveMessages { get; set; }
    }

    public enum PetitionKind
    {
        [Display(Name = "Загрязнение воздуха")]
        AirPollution,
        [Display(Name = "Загрязнение почвы")]
        SoilPollution,
        [Display(Name = "Добыча недр")]
        Excavation,
        [Display(Name = "Свалка")]
        Dump,
        [Display(Name = "Вырубка деревьев")]
        TreeCutting,
        [Display(Name = "Радиация")]
        Radiation
    }

    public enum PetitionStatus
    {
        Draft = 0,
        Sent = 100,
        Replied = 200,
        Success = 300,
        Failed = 500,
        Archived = 600
    }
}

File: GreenSignal.Mobile/app/src/main/java/com/example/greensignal/domain/model/response/Petition.kt
package com.example.greensignal.domain.model.response

import androidx.compose.runtime.toMutableStateList
import com.example.greensignal.data.remote.dto.response.PetitionDto
import com.example.greensignal.data.remote.dto.response.PetitionKind
import com.example.greensignal.data.remote.dto.response.PetitionStatus
import java.util.Date

data class Petition (
    val id: String = "",
    val serialNumber: String = "",
    val date: Date = Date(),
    val createdAt: Date = Date(),
    val departmentId: String = "",
    val department: Department = Department(),
    val incidentReportId: String? = null,
    val incidentReport: IncidentReport? = null,
    val parentPetitionId: String? = null,
    val parentPetition: Petition? = null,
    val inspectorId: String = "",
    val inspector: Inspector = Inspector(),
    val status: PetitionStatus = PetitionStatus.Draft,
    val kind: PetitionKind = PetitionKind.Dump,
    val attributeVersion: Int = 0,
    val attachements: MutableList<PetitionAttachment> = mutableListOf(),
    val attributes: MutableList<PetitionAttribute> = mutableListOf(),
    val receiveMessages: MutableList<Message> = mutableListOf()
)

fun PetitionDto.toPetition(): Petition {
    return  Petition(
        id = id,
        serialNumber = serialNumber,
        date = date,
        createdAt = createdAt,
        departmentId = departmentId,
        department = department.toDepartment(),
        incidentReportId = incidentReportId,
        incidentReport = incidentReport?.toIncidentReport(),
        parentPetitionId = parentPetitionId,
        parentPetition = parentPetition?.toPetition(),
        inspectorId = inspectorId,
        inspector = inspector.toInspectorAccount(),
        status = status,
        kind = kind,
        attributeVersion = attributeVersion,
        attachements = attachments.map { it.toPetitionAttachment() }.toMutableList(),
        attributes = attributes.map { it.toPetitionAttribute() }.toMutableList(),
        receiveMessages = receiveMessages.map { it.toMessage() }.toMutableList()
    )
}